3529/1/data_finder.rb

```diff
@@ -0,0 +1,91 @@
+class DataFinder
+  attr_reader :data_about_gem
+
+  def initialize(gem_name, url, client)
+    @data_about_gem = {}
+    @data_about_gem[:name] = gem_name
+    @url = url
+    @client = client
+    @repo_addr = adress_handle
+  end
+
+  def join_all_data
+    find_watchers_stars_forks
+    find_used_by_contributors_issues
+    @data_about_gem[:rate] = make_rate
+  end
+
+  private
+
+  def adress_handle
+    puts "Didn't find repository on github" unless @url
+    @repo_addr = if @url.include?('https://github.com/')
+                   @url.gsub('https://github.com/', '')
+                 else
+                   @url.gsub('http://github.com/', '')
+                 end
+  end
+
+  def find_watchers_stars_forks
+    find_watchers
+    find_stars
+    find_forks
+  end
+
+  def find_used_by_contributors_issues
+    find_used_by
+    find_contributers
+    find_issues
+  end
+
+  def make_rate
+    rate = find_watch_plus_starts + find_forks_plus_contributors
+    rate + @data_about_gem[:issues] * 0.05 + @data_about_gem[:used_by] * 0.5
+  end
+
+  def find_watch_plus_starts
+    @data_about_gem[:watched_by] * 0.15 + @data_about_gem[:stars] * 0.15
+  end
+
+  def find_forks_plus_contributors
+    @data_about_gem[:watched_by] * 0.15 + @data_about_gem[:stars] * 0.15
+  end
+
+  def find_forks
+    repo = @client.repo(@repo_addr)
```

Comment:
You repeat that line in a few methods(code duplication), that means that this line is good candidate for new method